app/controllers/groups_controller.rb

```diff
@@ -1,8 +1,13 @@
 class GroupsController < ApplicationController
+  load_and_authorize_resource
   before_action :set_group, only: [:show, :edit, :update, :destroy]
   before_action :authenticate_user!, :except => [:show, :index]
   def index
-    @groups = Group.all
+    @groups = Group.joins(:participants).where(
+      'owner_id = ? OR participants.user_id = ?',
+      current_user.id,
+      current_user.id
```

Comment:
`current_user.id` must be repeated in its parameters?

Reply:
i  did a refactor about this logic, now i only reference to current_user.id once in a scope method in Group model. Thanks for your CR